python/src/nanoarrow/_lib.pyx

```diff
@@ -166,6 +170,70 @@ cdef object alloc_c_array_view(ArrowArrayView** c_array_view):
     return PyCapsule_New(c_array_view[0], 'nanoarrow_array_view', &pycapsule_array_view_deleter)
 
 
+cdef void pycapsule_dlpack_deleter(object dltensor) noexcept:
+    cdef DLManagedTensor* dlm_tensor
+
+    # Do nothing if the capsule has been consumed
+    if PyCapsule_IsValid(dltensor, "used_dltensor"):
+        return
+
+    dlm_tensor = <DLManagedTensor*>PyCapsule_GetPointer(dltensor, 'dltensor')
+    if dlm_tensor == NULL:
+        PyErr_WriteUnraisable(dltensor)
+    # The deleter can be NULL if there is no way for the caller
+    # to provide a reasonable destructor
+    elif dlm_tensor.deleter:
+        dlm_tensor.deleter(dlm_tensor)
+
+
+cdef void view_dlpack_deleter(DLManagedTensor* tensor) noexcept with gil:
+    if tensor.manager_ctx is NULL:
+        return
+    PyMem_Free(tensor)
+    Py_DECREF(<CBufferView>tensor.manager_ctx)
+
+
+cpdef object view_to_dlpack(CBufferView view):
+    cdef DLManagedTensor* dlm_tensor = <DLManagedTensor*>PyMem_Malloc(sizeof(DLManagedTensor))
+
+    cdef DLTensor* dl_tensor = &dlm_tensor.dl_tensor
+    dl_tensor.data = <void*>view._ptr.data.data
+    dl_tensor.ndim = 1
+
+    cdef int64_t* _shape = <int64_t*>PyMem_Malloc(sizeof(int64_t))
```

Comment:
@paleolimbot another idea here is that we could use an existing value stored in the CBufferView, like `<int64_t*>(&view._shape)` (not needing a new tiny allocation). Except that this `_shape` attribute is marked as `Py_ssize_t`, which I assume is not guaranteed to be int64. Is there another attribute available? (eg the underlying nanoarrow C object? Although I don't see a length there, only the size in bytes)

Reply:
Just wanted to add that I tried using a `&view._shape` and it didn't work. So yes, great question!